### PR TITLE
Personal/v imazunder/migrate pr internal checks

### PR DESCRIPTION
**Pull request by v-imazumder**
## Description
Pipeline Path: Resolute/OSS
Repo: microsoft/fhir-server
Triggers: PR validation any branch

## Related issues
Addresses [issue #84623]Migrate "PR Internal Checks" Build Pipeline.

## Testing
Describe how this change was tested.

## FHIR Team Checklist
- [ ] **Update the title** of the PR to be succinct and less than 50 characters
- [ ] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with Azure API for FHIR if this will release to the Azure API for FHIR managed service (CosmosDB or common code related to service)
- [ ] Tag the PR with Azure Healthcare APIs if this will release to the Azure Healthcare APIs managed service (Sql server or common code related to service)
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: build/.vsts-PRInternalChecks-azureBuild-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,150 @@
+name: pr$(System.PullRequest.PullRequestNumber)-$(Date:yyyyMMdd)$(Rev:-r)
+parameters:
+- name: BuildConfiguration
+  default: release
+- name: NugetSecurityAnalysisWarningLevel
+  default: warn   
+variables:
+- name: vmImage
+  value: windows-2019
```
Comment (LTA-Thinking) [inline build/.vsts-PRInternalChecks-azureBuild-pipeline.yml:9]: Nit: Could this be changed to windows-latest. That way we don't need to update it in the future.

File: build/.vsts-PRInternalChecks-azureBuild-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      selectOrConfig: config
+      nugetConfigPath: nuget.config
+
+  - task: UseDotNet@2
+    displayName: Use .NET Core sdk
+    inputs:
+      inputs:
+      version: 5.0.x
+      useGlobalJson: true
+      installationPath: $(Agent.ToolsDirectory)/dotnet
```
Comment (LTA-Thinking) [inline build/.vsts-PRInternalChecks-azureBuild-pipeline.yml:45]: I think the only input needed here is the useGlobalJson: true. At least the second input: is not needed.

File: build/.vsts-PRInternalChecks-azureBuild-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    enabled: False
+    inputs:
+      FileDirPath: $(System.ArtifactsDirectory)
+
+  - task: PublishSecurityAnalysisLogs@2
+    displayName: Publish Security Analysis Logs
+    enabled: False
+    inputs:
+      ArtifactName: CodeAnalysisLogs - Antimalware
+...
```
Comment (LTA-Thinking) [inline build/.vsts-PRInternalChecks-azureBuild-pipeline.yml:150]: Nit: Could you remove the ...

File: build/.vsts-PRInternalChecks-azureBuild-pipeline.yml (hunk 1 above)
Comment (v-imazumder) [reply build/.vsts-PRInternalChecks-azureBuild-pipeline.yml:9]: Changed to windows-latest.

File: build/.vsts-PRInternalChecks-azureBuild-pipeline.yml [hunk 4]
```diff
@@ -40,9 +40,7 @@ jobs:
     displayName: Use .NET Core sdk
     inputs:
       inputs:
```
Comment (LTA-Thinking) [inline build/.vsts-PRInternalChecks-azureBuild-pipeline.yml:42]: Could you remove this second inputs: ?

**Review by LTA-Thinking — APPROVED** [review]

**Pull request merged**